src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/RunListenerImpl.java

```diff
@@ -61,67 +63,74 @@ public RunListenerImpl() {
 
     @Override
     public void onStarted(Run r, TaskListener listener) {
-        if (r instanceof AbstractBuild) {
-            AbstractBuild<?, ?> build = (AbstractBuild<?, ?>)r;
-            List<String> causes = new LinkedList<String>();
-            for (Object o : build.getCauses()) {
-                causes.add(o.getClass().getSimpleName());
-            }
+        List<String> causes = new LinkedList<String>();
+        for (Object o : r.getCauses()) {
+            causes.add(o.getClass().getSimpleName());
+        }
 
-            Cause.UpstreamCause upstreamCause = (Cause.UpstreamCause)r.getCause(Cause.UpstreamCause.class);
-            if (upstreamCause != null) {
-                causes.add(upstreamCause.getShortDescription());
+        Cause.UpstreamCause upstreamCause = (Cause.UpstreamCause)r.getCause(Cause.UpstreamCause.class);
```

Comment:
Iterate on r.getCauses() and pick up the Cause.UpstreamCause yourself. There can be several UpstreamCauses and get Cause(type) only returns one (we had this issue in the Downstream Stageview Plugin) .